### ISSUE TITLE
build script tries to install NTL in /usr/local/include/NTL

### ISSUE THREAD

**Issue by heiderich** [oscar-system/Singular.jl#45]
When executing `Pkg.build("Singular")`, the build script seems to try to install NTL under `/usr/local/include/NTL`, which usually fails when executed as a non-privileged user.

```
make[1]: Leaving directory '/home/florian/.julia/v0.6/Singular/deps/tmpEConDD/ntl-9.3.0/src'
touch all
mkdir -p -m 755 /usr/local/include
rm -rf /usr/local/include/NTL
mkdir -m 755 /usr/local/include/NTL
mkdir: cannot create directory ‘/usr/local/include/NTL’: Permission denied
makefile:427: recipe for target 'install' failed
make: *** [install] Error 1
===============================================[ ERROR: Singular ]================================================

LoadError: failed process: Process(`make install`, ProcessExited(2)) [2]
while loading /home/florian/.julia/v0.6/Singular/deps/build.jl, in expression starting on line 30

==================================================================================================================

=================================================[ BUILD ERRORS ]=================================================

WARNING: Singular had build errors.

 - packages with build errors remain installed in /home/florian/.julia/v0.6
 - build the package(s) and all dependencies with `Pkg.build("Singular")`
 - build a single package by running its `deps/build.jl` script

==================================================================================================================

```

**Comment by wbhart** [comment]
That's interesting. We do configure NTL with:

run(`./configure PREFIX=$vdir DEF_PREFIX=$nemovdir .....`)

and certainly neither of those directories is /usr/local/include/NTL. I'm
not sure what is causing this issue.

Perhaps we are issuing the configure command to NTL incorrectly.

**Comment by sebasguts** [comment]
According to the comments in NTL's `configure` script, this seems the right way to use it. Nevertheless, the `PREFIX` variable is ignored.

**Comment by wbhart** [comment]
I wonder if it is fixed in a later version of NTL, and if such a version
will work with Singular.

Otherwise, we might need to apply a patch.

I wonder if this could explain some of the difficulties we've had on OSX (I
realise it shouldn't work on Linux either, but maybe we've just been lucky).

**Comment by sebasguts** [comment]
This does not explain any of the difficulties I had on OS X, sorry. And this problem occurs on a Linux machine.

Singular itself works with newer versions of NTL, but I am not sure if this problem is fixed there.

**Comment by sebasguts** [comment]
TBH, I encountered this problem before, but was too lazy to dig into it. This is why I put the `sudo` here: https://github.com/oscar-system/OSCARDocker/blob/master/Dockerfile#L67 .

Should I investigate this?

**Comment by wbhart** [comment]
If you have time, that would be great.

Of course, adding sudo is not a way around it in general. Most users
wouldn't have sudo access to their machines. But it's understandable why
one would do it in a Docker image.

**Comment by sebasguts** [comment]
It basically solved it back then without much hassle for me, so I was fine with doing it in the image. I will investigate this.

**Comment by sebasguts** [comment]
So, after investigating this, I found the following explanation:

For correctly executing the configure script of ntl, you need a `libtool` binary in your path. This comes with the Ubuntu/Debian package `libtool-bin`, and not libtool. If you do not have that binary in your path, ntl's configure errors and does not overwrite ntl's already existing makefile.

Then the `build.jl` still executes `make` (which runs fine), and you get an error when `build.jl` tries to `make install` ntl.

I guess we should at least `try/catch` the configure call and see if configure was executed successfully. And then possibly provide a more elaborated error than ntl does by itself. I will implement this.

**Comment by heiderich** [comment]
Thank you for investigating this @sebasguts. After installing `libtool-bin` I could build this package successfully.

**Comment by fingolfin** [comment]
This issue was fixed in NTL 10.1.0, released 2016-10-14 -- perhaps you could switch from NTL 9.3.0 to that, or an even newer version of NTL?
